src/opm/output/eclipse/AggregateUDQData.cpp

```diff
@@ -64,30 +64,423 @@ namespace {
-        Opm::UDQTokenType::scalar_func_min,
-        Opm::UDQTokenType::binary_op_div
+            Opm::UDQTokenType::binary_op_add,
+            Opm::UDQTokenType::binary_op_sub
         };
+        for (const auto& tok_type : type_1) {
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
+        }
+        return type;
+    }
+
+    // function to return true if token is a binary operator: type
+    bool tokenTypeBinaryUnionOp(const Opm::UDQTokenType& token) {
+        bool type = false;
+        std::vector <Opm::UDQTokenType> type_1 = {
+            Opm::UDQTokenType::binary_op_uadd,
+            Opm::UDQTokenType::binary_op_umul,
+            Opm::UDQTokenType::binary_op_umin,
+            Opm::UDQTokenType::binary_op_umax
+        };
+        for (const auto& tok_type : type_1) {
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
+        }
+        return type;
+    }
+
+        // function to return true if token is a binary operator: type
+    bool tokenTypeParen(const Opm::UDQTokenType& token) {
+        bool type = false;
+        std::vector <Opm::UDQTokenType> type_1 = {
+            Opm::UDQTokenType::open_paren,
+            Opm::UDQTokenType::close_paren,
+        };
+        for (const auto& tok_type : type_1) {
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
+        }
+        return type;
+    }
+
+    // function to return true if token is a binary operator: anytype
+    bool tokenTypeBinaryOp(const Opm::UDQTokenType& token) {
+        bool type = false;
+        std::vector <Opm::UDQTokenType> type_1 = {
+            Opm::UDQTokenType::binary_op_add,
+            Opm::UDQTokenType::binary_op_sub,
+            Opm::UDQTokenType::binary_op_div,
+            Opm::UDQTokenType::binary_op_mul,
+            Opm::UDQTokenType::binary_op_pow,
+            Opm::UDQTokenType::binary_op_uadd,
+            Opm::UDQTokenType::binary_op_umul,
+            Opm::UDQTokenType::binary_op_umin,
+            Opm::UDQTokenType::binary_op_umax,
+            Opm::UDQTokenType::binary_cmp_eq,
+            Opm::UDQTokenType::binary_cmp_ne,
+            Opm::UDQTokenType::binary_cmp_le,
+            Opm::UDQTokenType::binary_cmp_ge,
+            Opm::UDQTokenType::binary_cmp_lt,
+            Opm::UDQTokenType::binary_cmp_gt
 
-        int num_type_1 = 0;
+        };
         for (const auto& tok_type : type_1) {
-            num_type_1 += tokens.count(tok_type);
+            if (token == tok_type) {
+                type = true;
+                break;
+            }
         }
-        type = (num_type_1 > 0) ? -1 : -4;
         return type;
     }
 
+    // function to return the precedence of the current operator/function
+    bool operatorToken(const Opm::UDQTokenType& token) {
+        bool opTok = false;
+        if (tokenTypeFunc(token)) opTok = true;
```

Comment:
Looks like it should be possible to turn this function around;
```
bool opTok = true;
if (...) opTok = false;
return opTok;
```
?

Reply:
The logic used is correct.